nsdl-c/sn_nsdl_lib.h

```diff
@@ -146,7 +146,7 @@ typedef struct sn_nsdl_static_resource_parameters_ {
     uint16_t    resourcelen;                /**< 0 if dynamic resource, resource information in static resource */
     bool        external_memory_block:1;    /**< 0 means block messages are handled inside this library,
                                                  otherwise block messages are passed to application */
-    uint8_t     mode:2;                     /**< STATIC etc.. */
+    int         mode:2;                     /**< STATIC etc.. */
```

Comment:
Never use "int" bitfields. It's undefined whether they're signed or unsigned. Always make them explicitly "signed" or "unsigned" (you can then leave out the "int").

GCC and ARMCC do actually have different behaviour on this - been bitten by that before.